Src/IronPython.Modules/mmap.cs

```diff
@@ -417,7 +417,7 @@ void CheckFileAccessAndSize(Stream stream) {
                             throw PythonOps.OSError(PythonExceptions._OSError.ERROR_ACCESS_DENIED, "Invalid access mode");
                         }
 
-                        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
+                        if (!isWindows) {
```

Comment:
Not really an issue since we're not calling native APIs here, but any idea if the platform analyzer would pick up on an `isWindows` variable and guard against Windows calls?

Reply:
As it is written now, no platform-specific calls are allowed in `CheckFileAccessAndSize`, regardless whether they are conditioned by `isWindows` in any way. .NET has `SupportedOSPlatformGuardAttribute`, but it cannot be applied to a parameter.

The reason why I replaced the platform check with a parameter is because having the platform test in this function was invalidating the platform context after the first call site, and I needed to make a call to `PythonNT.dupUnix`.

Reply:
Cool, didn't know (or forgot) that `SupportedOSPlatformGuardAttribute` existed.  Weird that is breaks the platform check. Alternatively the parameter could be called `allowGrow` or something.